### PR TITLE
fix+feat: fix infinte-loading due to non-call of callback and expose error message into frontend

### PR DESCRIPTION
**Pull request by ilsubyeega**
## Summary

When a new user tried to sign up with social login, while registration disabled, the callback won't being called and would have to wait for a long time due to no callbacks. Also This PR also introduces append `error` parameter to `/login` frontend route to show the error of oauth.

- Backend
  - return callback if registration has been closed.
  - Wrap `passport.authenticate` to handle the error message and then redirect into.
  - Expose error message by redirecting to `/login?error=<MESSAGE>`.

- Frontend
  - `Login.tsx`: Extract `?error` parameters from the URL and show it as an error message.
    - created `customError` state since `error` state always localized.
    - duplicated error element for `customError` usage.

- Concerns
  - `customError` state is duplicated with `error` state. And might need to change names to avoid confusion.

Since I'm not used on express.js and React, Please let me know if i wrong of this

## Change Type

Please delete any irrelevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Translation update
- [ ] Documentation update

## Testing

Please describe your test process and include instructions so that we can reproduce your test. If there are any important variables for your testing configuration, list them here.

### **Test Configuration**:

I have no idea which unit test should i create.

Tested with `ALLOW_SOCIAL_LOGIN=true` and `ALLOW_SOCIAL_REGISTRATION=false` env.

## Checklist

Please delete any irrelevant options.

- [ ] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [ ] I have commented in any complex areas of my code
- [ ] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [ ] I have written tests demonstrating that my changes are effective or that my feature works
- [ ] Local unit tests pass with my changes
- [ ] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

**Comment by danny-avila** [comment]
thanks! looks great, will test it and merge by end of weekend

File: api/server/routes/oauth.js [hunk 1]
```diff
@@ -14,6 +14,21 @@ const domains = {
...
 router.use(loginLimiter);
 
+const authenticate = (strategy, options) => {
+  return (req, res, next) =>
+    passport.authenticate(strategy, options, (err, _user, info) => {
+      if (err) {
+        return next(err);
+      }
+
+      if (info) {
```
Comment (danny-avila) [inline api/server/routes/oauth.js]: Thanks again for the PR. Attempting login without the environment variables set, it looks like `info` is always defined and returns with redirect and the empty object: 
image

Tested with google, github, and discord, all would include an empty info object which generates the `[object Object]` string.

You could handle the edge case of the object string, as well as make sure this condition only happens when it's truly an error.

**Review by danny-avila — CHANGES_REQUESTED** [review]
It does work as expected when social registration is disabled and attempting to use a new account. However, the issue exists when using an existing account.

image

File: api/server/routes/oauth.js (hunk 1 above)
Comment (ilsubyeega) [reply api/server/routes/oauth.js]: I've tested the case of an existing user logging in now, I would appreciate it if you check it again.

File: client/src/components/Auth/Login.tsx [hunk 2]
```diff
@@ -136,6 +146,14 @@ function Login() {
             {localize(getLoginError(error))}
           </div>
         )}
+        {customError && (
```
Comment (danny-avila) [inline client/src/components/Auth/Login.tsx]: Just needs merge conflict resolved and I think this is good to go!

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: client/src/components/Auth/Login.tsx (hunk 2 above)
Comment (ilsubyeega) [reply client/src/components/Auth/Login.tsx]: Aight I'm okay with this!

**Pull request closed without merging**